ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/blockimport/BlockImporter.java

```diff
@@ -37,32 +40,24 @@ public BlockImporter(ChainStorageClient storageClient, EventBus eventBus) {
     this.eventBus = eventBus;
   }
 
-  /**
-   * @param block The block to check
-   * @return {@code true} If this block is from the future according to our node's clock
-   */
-  public boolean isFutureBlock(BeaconBlock block) {
-    if (storageClient.isPreGenesis()) {
-      return true;
-    }
-    final UnsignedLong genesisTime = storageClient.getStore().getGenesisTime();
-    return ForkChoiceUtil.isFutureBlock(block, genesisTime, storageClient.getStore().getTime());
-  }
-
-  /**
-   * @param block The block to check.
-   * @return {@code true} If block is attached to the local chain
-   */
-  public boolean isBlockAttached(BeaconBlock block) {
-    return storageClient.getBlockState(block.getParent_root()).isPresent();
-  }
-
-  public void importBlock(BeaconBlock block) throws StateTransitionException {
+  public BlockImportResult importBlock(BeaconBlock block) {
```

Comment:
Probably worth adding a `@javax.annotation.CheckReturnValue` to this so that compilation fails if we ignore the result.